addons/hr_expense/models/hr_expense.py

```diff
@@ -243,6 +244,10 @@ def _compute_attachment_number(self):
         for expense in self:
             expense.attachment_number = attachment.get(expense._origin.id, 0)
 
+    def _compute_attachment_ids(self):
+        for expense in self:
+            expense.attachment_ids = self.env['ir.attachment'].search([('res_model', '=', 'hr.expense'), ('res_id', '=', expense.id)])
```

Comment:
It's forbidden to do a aggregate on the ids of the attachments by the ORM. (I presume it's for some security reason, or to not bypass the ACL).

I've tried initially to do a `_read_group` and array_agg(id) and was met with the error.

I know it's not ideal, but if I can't agg on ids, I don't think I can do better? Unless I am missing something?

Reply:
Isn't this equivalent to a One2many with a domain, or I miss something ?
```python
attachment_ids = fields.One2many('ir.attachment', 'res_id', domain=[('res_model', '=', 'hr.expense')], string="Attachments")
```
(It will be batched automatically by the ORM)